Sources/Swift/Core/MetricKit/SentryMXCallStackTree.swift

```diff
@@ -1,3 +1,4 @@
+@_implementationOnly import _SentryPrivate
```

Comment:
`m`: I feels a bit strange to me now that we have a dependency on `_SentryPrivate` here. This is now a bit mixed. We have both MetricKitCallStack JSON parsing with the `JSONDecoder` and also `toDebugMeta` and `public func prepare(event: Event, inAppLogic: SentryInAppLogic?, handled: Bool) `.

I would prefer keeping this file focused on only parsing JSON data. I would move the conversion logic to a different file. It can be an extension, but doing this in a different file feels a bit cleaner to me.